ui5/browser/controller/Browser.controller.js

```diff
@@ -142,54 +163,191 @@ sap.ui.define(['sap/ui/core/Component',
+
+      /* ========================================================= */
+      /* =============== Generic factory functions =============== */
+      /* ========================================================= */
+
+      getElementFromCurrentTab: function(element) {
+         const currentTabID = this.getView().byId("myTabContainer").getSelectedItem();
+         return sap.ui.getCore().byId(currentTabID + element);
+      },
+
+      /* ========================================================= */
+      /* =============== Generic factory functions =============== */
+      /* ========================================================= */
+
+      /* =========================================== */
+      /* =============== Code Editor =============== */
+      /* =========================================== */
+
+      newCodeEditor: async function () {
+         const oTabContainer = this.getView().byId("myTabContainer");
+
+         const oTabContainerItem = new TabContainerItem("CodeEditor" + this.globalId, {
+            icon: "sap-icon://write-new-document",
+            name: "Code Editor",
+            additionalText: "untitled"
          });
 
-         // TODO: use proper openui5 methods to get aggregation
-         let defaultCodeEditor = this.getView().byId("defaultCodeEditor").getContent()[0].mAggregations.contentAreas[1];
-         defaultCodeEditor.setModel(new JSONModel({
+         const oCodeEditorFragment = this.newCodeEditorFragment(this.globalId);
+         let ID = this.globalId;
+         this.globalId++;
+
+         oTabContainerItem.addContent(oCodeEditorFragment);
+         oTabContainer.addItem(oTabContainerItem);
+         oTabContainer.setSelectedItem(oTabContainerItem);
+
+         sap.ui.getCore().byId("CodeEditor" + ID + "Editor").setModel(new JSONModel({
             code: "",
             ext: "",
             filename: "",
             fullpath: "",
             modified: false
          }));
+      },
 
-         let splitterUpperContent = this.getView().byId("defaultCodeEditor").getContent()[0].mAggregations.contentAreas[0].getContent();
-         splitterUpperContent[0].attachChange(this.onChangeFile, this);
-         splitterUpperContent[1].attachPress(this.onSaveAs, this);
-         splitterUpperContent[2].attachPress(this.onSaveFile, this);
-         splitterUpperContent[3].attachPress(this.onRunMacro, this);
-
-         defaultCodeEditor.attachChange( function() {
+      newCodeEditorFragment: function (ID) {
+         let fragment = new Splitter("CodeEditor" + ID + "Splitter", {
+            orientation: "Vertical",
+            contentAreas: [
+               new Toolbar("CodeEditor" + ID + "Toolbar", {
+                  content: [
+                     new FileUploader("CodeEditor" + ID + "FileUploader", {}),
+                     new Button("CodeEditor" + ID + "SaveAs", {
+                        text: "Save as...",
+                        tooltip: "Save current file as...",
+                     }),
```

Comment:
You are using ID to assign event handlers afterwards.
Is it not simpler just assign handlers when Button is created like:
`new Button(...).attachPress(this.onPressFunc, this)`

Reply:
Yes, you are right, but i found an other better way, thanks to you, which is 
```
new Button("CodeEditor" + ID + "Save", {
                        text: "Save",
                        tooltip: "Save current file",
                        press: [this.onSaveFile, this]
                     }),
```

Reply:
Yes, this is even better solution!

Reply:
But now please remove ID assignment for objects where it does not used.
It should be general rule - assign ID only when it is required

Reply:
Ok, will do, we can still re assign them if needed after